lib/birthday_bot.rb

```diff
@@ -3,26 +3,45 @@
 require 'birthday_reader'
 
 class BirthdayBot 
-  @@path = "birthdays.txt"
 
   def initialize()
     @config = ConfigReader.new
     @config.load('configurations.json')
   end
 
   def start!
-    birthdays = BirthdayReader.get_birthdays(@@path)
-    today = Time.now 
+    birthdays = BirthdayReader.get_birthdays(@config.birthdays_path)
 
-    puts "Checking who was born today (#{today.to_s})"
-    birthdays.each do |b|
-      if (b[3].to_i == today.month) && (b[4].to_i == today.day)
-        message = "#{@config.greeting_message} #{b[0]} #{b[1]}" 
-        HTTParty.post(@config.slack_url, body: { channel: @config.channel_name,
-                                                 username: @config.bot_name,
-                                                 text: message,
-                                                 icon_emoji: @config.bot_emoji }.to_json)
+    puts "Checking who was born today ( #{ Time.now.to_s } )"
+    unless birthdays.nil?
+      users = "#{ mention birthdays[0] }"
+      if birthdays.count > 1
+        puts "#{ birthdays.count } people were born today"
+        if birthdays.count == 2
+          users = " #{ mention birthdays[0] } and #{ mention birthdays[1] } "
+        else
+           for i in 1..birthdays.count-2
+             users.concat( ", #{ mention birthdays[i] }" )
+           end
+           users.concat( " and #{ mention birthdays[i+1] } " )
+        end
       end
+      message = "#{users} #{@config.greeting_message}"
+      HTTParty.post(@config.slack_url, body: { channel: @config.channel_name,
+                                               username: @config.bot_name,
+                                               text: message,
+                                               icon_emoji: @config.bot_emoji }.to_json)
+    else
+      puts "Today is a day that no one was born"
     end
```

Comment:
Have you looked at my previous comment with a suggestion to refactor this function ?

```
puts "#{birthdays.count} were born today"

today = Time.now
birthdays_today = birthdays[today.month][today.day]

unless birthdays_today.empty?
  users = build_user_list(birthdays_today)
  message = "#{users} #{@config.greeting_message}"
  HTTParty.post(@config.slack_url, body: { channel: @config.channel_name,
    username: @config.bot_name,
    text: message,
    icon_emoji: @config.bot_emoji }.to_json)
end
 
# ...

def build_user_list(birthdays)
  if birthdays.size == 1
    birthdays.first
  else
    users = birthdays.take(birthdays.count - 1).join(", ")
    users += " and #{birthdays[birthdays.count - 1]}" if birthdays.count > 1
  end
end
```